Updates to 'Using Postman' doc per Issue #696

cjjaramillo opened the pull request:
https://github.com/odpi/egeria-docs/issues/696

I've added an alternate way to import the collections needed for the dojo using links to import instead of authenticating through github and removed the previous procedure as it is currently not working because due to Postman.

davidradl commented on site/docs/education/tutorials/postman-tutorial/overview.md:
````diff
@@ -45,24 +45,77 @@ postman files later, you can just create a new environment again to work with
+    https://raw.githubusercontent.com/odpi/egeria/40ec581a8dbf82eeb7e96e2412099358193e8fc1/open-metadata-implementation/admin-services/Egeria-admin-services-operational.postman_collection.json
+    ```
+
+* Confirm the correct collection will be imported
````
how should the user do this . Maybe add by pressing a particular button

planetf1 replied:
In this case it is by seeing the new collection is imported (4th image in sequence). There's no button. It's the response after importing. I think if the image was closely associated with that last confirm text it would be clear?

I'm not generally a big fan of collapsible section - but I think here it would work well as per David's comment above. The  screenshot could be directly with each of these bullet points.  My reasoning is that perhaps many people don't need the screenshots, just the clear text, and if they do want to see they can expand out to view in more detail?

cjjaramillo replied:
I've given more detailed instructions, and included the screenshot in a collapsible section.

davidradl commented on site/docs/education/tutorials/postman-tutorial/overview.md:
````diff
@@ -45,24 +45,77 @@ postman files later, you can just create a new environment again to work with
+    ```
+
+* Confirm the correct collection will be imported
+* Confirm the collection was imported successfully
````
how should the user do this?

cjjaramillo replied:
I've given more detailed instructions, and included the screenshot in a collapsible section.

davidradl commented on site/docs/education/tutorials/postman-tutorial/overview.md:
```diff
@@ -27,7 +27,7 @@ Install Postman from [the Postman downloads site :material-dock-window:](https:/
 Once Postman is installed, start up the application.  You should see an initial page something like this:
 
-![Postman main window](postman-mainmenu.png)
+![Postman main window](postman-main-menu.png)
```
I suggest we should mention what platform and version of postman you have taken the screen shots. e.g. on a Mac M1 postman version ... this should match the postman version mentioned earlier.

cjjaramillo replied:
I've added this prior to displaying the first screenshot.

davidradl commented on site/docs/education/tutorials/postman-tutorial/overview.md:
```diff
@@ -45,24 +45,77 @@ postman files later, you can just create a new environment again to work with
-![Creating a new Environment 1](postman-workspace1.png)
-![Creating a new Environment 2](postman-workspace2.png)
-![Creating a new Environment 3](postman-workspace3.png)
+![Creating a new Environment 1](postman-create-workspace1.png)
```
I wonder if it would be easier to read if it were done like the installing Intellij where each screen shot it sin a folding section with a description, it is then more obvious what each screen shot means. I guess this would be easier for a screen reader which could not see the screen shot as well.

cjjaramillo replied:
I've given more detailed instructions, and included the screenshot in a collapsible section.

davidradl commented on site/docs/education/tutorials/postman-tutorial/overview.md:
```diff
@@ -45,24 +45,77 @@ postman files later, you can just create a new environment again to work with
-![Creating a new Environment 2](postman-workspace2.png)
-![Creating a new Environment 3](postman-workspace3.png)
+![Creating a new Environment 1](postman-create-workspace1.png)
+![Creating a new Environment 2](postman-create-workspace2.png)
```
I wonder if we could use the figure and figcaption https://squidfunk.github.io/mkdocs-material/reference/images/

the png name refers to workspace2 and the last one workspace1 which does not seem to reflect the content
As it implies there are 2 workspaces

cjjaramillo replied:
I've used captions for the screenshots that are not explained in a collapsible section.

davidradl commented on site/docs/education/tutorials/postman-tutorial/overview.md:
```diff
@@ -45,24 +45,77 @@ postman files later, you can just create a new environment again to work with
+![Import Begin](postman-collection-import-begin.png)
+![Select import source](postman-collection-url.png)
+![Confirm Collection to import](postman-collection-importconf.png)
+![Confirm imported successfully](postman-collection-importcomplete.png)
```
i suggest putting each of these text lines next to the appropriate screen shot, so the contecxt of the screen shots is more obvious. Maybe using the twisties like in the Intellij section.

cjjaramillo replied:
I've given more detailed instructions, and included the screenshot in a collapsible section.

davidradl commented on site/docs/education/tutorials/postman-tutorial/overview.md:
````diff
@@ -45,24 +45,77 @@ postman files later, you can just create a new environment again to work with
+    https://raw.githubusercontent.com/odpi/egeria/40ec581a8dbf82eeb7e96e2412099358193e8fc1/open-metadata-resources/open-metadata-samples/postman-rest-samples/Egeria.postman_environment.json
+    ```
+
+* Confirm the environment to be imported
````
how?

cjjaramillo replied:
I've given more detailed instructions, and included the screenshot in a collapsible section.

davidradl commented on site/docs/education/tutorials/postman-tutorial/overview.md:
````diff
@@ -45,24 +45,77 @@ postman files later, you can just create a new environment again to work with
+    ```
+
+* Confirm the environment to be imported
+* Confirm the environment was imported successfully
````
how

cjjaramillo replied:
I've given more detailed instructions, and included the screenshot in a collapsible section.

davidradl commented on site/docs/education/tutorials/postman-tutorial/overview.md:
```diff
@@ -45,24 +45,77 @@ postman files later, you can just create a new environment again to work with
+* Confirm the environment to be imported
+* Confirm the environment was imported successfully
+
+![No existing environments](postman-environments-empty.png)
```
I suggest having text and twisties with these screenshots as well

cjjaramillo replied:
I've given more detailed instructions, and included the screenshot in a collapsible section.

davidradl commented on site/docs/education/tutorials/postman-tutorial/overview.md:
```diff
@@ -45,24 +45,77 @@ postman files later, you can just create a new environment again to work with
+![Confirm Collection to import](postman-collection-importconf.png)
+![Confirm imported successfully](postman-collection-importcomplete.png)
+
+* Repeat the same procedure for the following Egeria collections:
```
Could we display these as a table with a column for the collection name an one for the url? If this formats poorly these we should not , but it would be worth playing with.

planetf1 replied:
A table could work nicely.

Trying to make sure the links are easily copyable would be helpful. it reduces the mouse movement etc to select.
I offered one option (which implies a quoted file) above, any ideas on more?
I think that speed of being able to copy is more important than exact format, as user has to do this a number of times

cjjaramillo replied:
I've been trying the table suggestion. I can set up the table however, I've tried a bunch of different approaches and from what I can find, there isn't a way to nest a "copy url" option in the table (cell content can only be on one line). I was thinking to abandon the idea of a table and instead use something like this instead: 
<img width="857" alt="Screenshot 2023-02-28 at 11 10 09 AM" src="https://user-images.githubusercontent.com/25217353/221911974-51e4c4f9-c707-4081-8486-fad9e274427b.png">

Do you think this is an acceptable approach?

cjjaramillo replied:
I tweaked my previous idea a bit. Instead of a collapsible section as seen above, I've given each its own section that is not collapsible since the URLs will be required. The URLs have the copy link icon as well. The collapsible sections were really only used for optional screenshots should the user want to reference.

davidradl commented on site/docs/education/tutorials/postman-tutorial/overview.md:
```diff
@@ -85,9 +138,9 @@ Egeria platform. Tutorial content will walk you through this explicitly.
-![Select Egeria environment](postman-env-select.png)
-![Edit Egeria environment](postman-env-editselect.png)
-![View and edit environment values](postman-env-edit.png)
+![Select Egeria environment](postman-env-switch.png)
```
these screen shots should be associated with text and maybe in a twisty (section)

cjjaramillo replied:
I've given more detailed instructions, and included the screenshot in a collapsible section.

davidradl commented on site/docs/education/tutorials/postman-tutorial/overview.md:
```diff
@@ -73,8 +126,8 @@ turn off certificate validation in postman:
 * Go to settings
 * Ensure 'SSL certificate verification' is switched off
```
again I would associate the text and screenshot in a section

cjjaramillo replied:
I've given more detailed instructions, and included the screenshot in a collapsible section.

planetf1 commented on site/docs/education/tutorials/postman-tutorial/overview.md:
````diff
@@ -45,24 +45,77 @@ postman files later, you can just create a new environment again to work with
+* Paste the URL found below for the `Egeria-admin-services-operational` collection
+
+    ```
+    https://raw.githubusercontent.com/odpi/egeria/40ec581a8dbf82eeb7e96e2412099358193e8fc1/open-metadata-implementation/admin-services/Egeria-admin-services-operational.postman_collection.json
````
I'm thinking how we make this easier. One option is to change these links into something like:
<img width="1209" alt="Screenshot 2023-02-28 at 09 34 59" src="https://user-images.githubusercontent.com/7292002/221812383-c6d3a939-32fb-403e-a28c-7fbe33944032.png">

This will enable a 'copy' link 

Unfortunately the button overlaps with the text, and the .url filetype here is just made up. I will look to see if there's a better way.

<img width="794" alt="Screenshot 2023-02-28 at 09 34 15" src="https://user-images.githubusercontent.com/7292002/221812400-8647ddcf-6170-440d-8681-f89ed5faa83c.png">

This is useful though, since a triple click on the URL will result in
<img width="779" alt="Screenshot 2023-02-28 at 09 37 53" src="https://user-images.githubusercontent.com/7292002/221813304-30eed26b-1fd7-4a0a-b1bf-4dde5391fe30.png">

cjjaramillo replied:
I tried to find other approaches, but your suggestion seemed to be the best. I used this through the document when listing a URL.

planetf1 reviewed (commented):
Thanks for making the update

Trying to make the link more clickable would be useful.

When I put in the original screenshots we ended up with a mix of dark/light (updates were done last minute before a dojo run).

Getting them consistent is a great step - thanks.
I do wonder if longer term they'd be better with a light background - easier to read when small, but that's a distinct change, so we wouldn't have to not merge just for that improvement.

planetf1 commented:
Additional note - we use mkdocs with the material plugin for our formatting.
Not all features are enabled, and we use the community version, but the docs at https://squidfunk.github.io/mkdocs-material/reference/ can be useful to look for the best formatting options -> https://squidfunk.github.io/mkdocs-material/reference/

cjjaramillo commented:
I've added individual steps - each in its own collapsible section that includes a screenshot. I did not add captions to the images in the collapsible sections since the images really just support the text they are included with. I added captions to the images that are not contained in a collapsible section. Please let me know if any other improvements are needed.

planetf1 reviewed (changes requested):
I think that extra commits/file changes have been inadvertently added.
Recommend editing the PR to remove
(new branch, cherry pick may be simplest)

davidradl commented:
@cjjaramillo I assume we can close this now?
